express-notes-api/index.js

```diff
@@ -0,0 +1,102 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+const fs = require('fs');
+
+app.get('/api/notes', (req, res) => {
+  const newArr = [];
+  for (const x in data.notes) {
+    newArr.push(data.notes[x]);
+  }
+  res.json(newArr);
+});
+app.get('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  if (Number.isInteger(id) && id > 0) {
+    if (!data.notes[id]) {
+      res.status(404).send({ error: `cannot find note with id ${id}` });
+    } else {
+      res.status(200).json(data.notes[id]);
+    }
+  }
```

Comment:
Here you need an `else` so that if the number is not an integer or is less than `0` an error message will be sent back. As it is now if the `id` doesn't checkout the server will "hang" and a response will never be sent back to the client.
```JAVASCRIPT
} else {
  res.status(400).json({ error: 'Id must be a positive integer' })
}
```